portal/static/js/coredata.js

```diff
@@ -1,54 +1,152 @@
+            var ethnicityData = this.getEthnicityData();
+            var demoArray = {resourceType: "Patient", extension: []};
+            if (raceData) {
+                demoArray.extension.push(raceData);
+            }
+            if (ethnicityData) {
+                demoArray.extension.push(ethnicityData);
+            }
+            tnthAjax.putDemo(this.subjectId, demoArray);
+        };
+        this.getDemoData = function() {
+            if (!this.demoFieldsEnabled()) {
+                return;
+            }
+            tnthAjax.getDemo(this.subjectId);
+        };
+        this.initDemoFieldEvents = function() {
+            if (!this.demoFieldsEnabled()) {
+                return;
+            }
+            var self = this;
+            this.getRaceFields().on("click", function() {
+                self.setDemoData();
+            });
+            this.getEthnicityFields().on("click", function() {
+                self.setDemoData();
+            });
+        };
+        this.initContinueButtonEvent = function() {
+            // Class for both "done" and "skip" buttons
+            var self = this;
+            $(".continue-btn").on("click", function(event){
+                event.preventDefault();
+                $(this).attr("disabled", true);
+                $(".loading-indicator").show();
+                try {
+                    window.location.replace(self.getReturnAddress());
+                } catch(e) {
+                    self.setError(i18next.t("Error occurred when redirecting to destination url"));
+                    //report error if invalid return address is used here
+                    tnthAjax.reportError(self.subjectId, self.getAPIUrl(), e.message, true);
+                    $(this).attr("disabled", false);
+                    $(".loading-indicator").hide();
+                    $(".error-continue").text(e.message);
+                }
+            });
+        };
+        this.initFieldEvents = function() {
+            this.initDemoFieldEvents();
+            this.initContinueButtonEvent();
+        };
+        this.getReturnAddress = function() {
+            return $("#procReturnAddress").val() || "/";
+        };
+        this.getAPIUrl = function() {
+            return $("procAPIUrl").val() || "/api/coredata/acquire";
+        };
+        this.setError = function(message) {
+            $("#coreDataError").html(message || "");
+        };
+        this.setFooter = function() { //only for GIL footer
+            var self = this;
+            tnthAjax.setting("GIL", this.subjectId, false, function(data) {
+                if (data && data.GIL) {
+                    tnthAjax.getPortalFooter(self.subjectId, false, "core_data_footer");
+                }
+            });
+        };
+    };
+    $(document).ready(function(){
+        (new CoreDataObj()).init();
     });
-    tnthAjax.getPortalFooter(subjectId, false, "core_data_footer");
-});
+})();  /*eslint wrap-iife: off */
```

Comment:
If I understand correctly, this disables the `wrap-iife` linting rule for this entire function...

PS some notes about [IIFE style](http://airbnb.io/javascript/#functions--iife)